packages/ppa-nest-api/src/decorators/api-created-entity-response.decorator.ts

```diff
@@ -0,0 +1,26 @@
+import { Type, applyDecorators } from '@nestjs/common';
+import {
+  ApiCreatedResponse,
+  ApiExtraModels,
+  getSchemaPath,
+} from '@nestjs/swagger';
+import { SingleEntityResponseDto } from 'src/util/single-entity-response.dto';
+
+export const ApiCreatedEntityResponse = <T extends Type<unknown>>(DTO: T) =>
```

Comment:
sorry I don't know much about nestjs can you please explain me why is this for?

Reply:
There's an specification for describing RESTful APIs in a standard format, it's called OpenAPI and it provides a machine-readable way to define API endpoints, parameters, responses, and more.
There's another thing called Swagger, which is like a set of tools that use this specification to generate documentation, among other things...

When you launch the Nestjs app and navigate to `http://localhost:3000/api#/` you'll find this documentation page about the endpoints:

<img width="490" alt="image" src="https://github.com/mauro1998/Labs-Web-Experiment/assets/20090908/7d555688-f6ff-4b93-ba1b-a218b26f3bdd">

Nestjs provides an integration for Swagger that allows you to generate this documentation using Typescript types and Decorators. There are some built-in Decorators for describing an endpoint, for example:

```
@Post()
@ApiBadRequestResponse() <--- built-in decorator to indicate that this post request may return 400 error
@ApiInternalServerErrorResponse() <--- or 500 error
@ApiCreatedResponse() <--- or 201 ok generic response
async create(@Body() createSessionDto: CreateSessionDto) {
  ...
}
```

So what I did here was just to create a custom decorator that extends the `ApiCreatedResponse` decorator to provide additional documentation about the `POST` endpoint:

```
@Post()
@ApiCreatedEntityResponse(SessionEntity) <--- 201 status code + the response model
async create(@Body() createSessionDto: CreateSessionDto) {
  ...
}
```

So that when Swagger reads this, it will generate the schema of the response in the docs:

<img width="646" alt="image" src="https://github.com/mauro1998/Labs-Web-Experiment/assets/20090908/1476a67c-9653-4f7e-90b5-c76d8292542a">